src/db.ts

```diff
@@ -231,19 +234,97 @@ export class Database {
       return null;
     }
 
-    const upstreamPRIDs = (result.upstream_pr_ids as string).split(",").map((idStr) => parseInt(idStr));
-
     return {
       repo: repo,
       branch: branch,
       upstreamRepo: upstreamRepo,
-      upstreamPRIDs: upstreamPRIDs,
+      upstreamPRIDs: (result.upstream_pr_ids as string).split(",").map((idStr) => parseInt(idStr)),
       action: result.action,
       prID: result.pr_id,
       githubIssue: result.github_issue,
     };
   }
 
+  /**
+   * Get a pending pull-request (PR) with the input GitHub issue ID.
+   * @param {Repo} repo the repo that the GitHub issue belongs to.
+   * @param {number} issueID the GitHub issue ID (technically the "number") to search for.
+   * @return {Promise<PendingPR | null>} a Promise that resolves to the PendingPR object or null if it's not found.
+   */
+  async getPendingPRByIssue(repo: Repo, issueID: number): Promise<PendingPR | null> {
```

Comment:
I think the doc here needs to explain better the difference between this and `getPendingPRByPRID` - I think this one find a PR that is linked to an issue which the bot opened because a PR couldn't be automatically merged, is that right? Since (in my understanding) issue numbers and PR numbers are effectively the same in GitHub, the doc is a little confusing right now.